Source/WebGPU/WebGPU/RenderPipeline.h

```diff
@@ -58,16 +59,24 @@ class RenderPipeline : public WGPURenderPipelineImpl, public RefCounted<RenderPi
     bool isValid() const { return m_renderPipelineState; }
 
     id<MTLRenderPipelineState> renderPipelineState() const { return m_renderPipelineState; }
+    MTLPrimitiveType primitiveType() const { return m_primitiveType; }
+    std::optional<MTLIndexType> maybeIndexType() const { return m_indexType; }
+    MTLWinding frontFace() const { return m_frontFace; }
+    MTLCullMode cullMode() const { return m_cullMode; }
```

Comment:
I kind of think the getters shouldn't get internal Metal types, but instead should get the WebGPU type, and we can add a helper in a utilities file to convert. Returning Metal types feels kind of like a layering violation.

Reply:
This is easy for `primitiveType`, `frontFace`, and `cullMode`, but I don't know if its so simple for `renderPipelineState` which is used in the RenderPassEncoder. 

Do you see a feasible way to achieve this for `renderPipelineState`? I will also see if I can break the coupling somehow between RenderPipeline and RenderPassEncoder